### PR TITLE
Add android flavor to dartDefines.

### PR DESCRIPTION
**Pull request by GChanathip**
this PR fix issue https://github.com/leancodepl/patrol/issues/2256

### CODE REVIEW

File: packages/patrol_cli/lib/src/crossplatform/app_options.dart [hunk 1]
```diff
@@ -115,14 +115,22 @@ class AndroidAppOptions {
     final target = '-Ptarget=${flutter.target}';
     cmd.add(target);
 
+    // Create modifiable Map
+    final effectiveDartDefines = Map<String, String>.of(flutter.dartDefines);
+
+    // Add flavor to dart defines
+    if (flutter.flavor != null) {
+      effectiveDartDefines['FLUTTER_APP_FLAVOR'] = flutter.flavor!;
+    }
```
Comment (piotruela) [inline packages/patrol_cli/lib/src/crossplatform/app_options.dart:124]: Nit:
```suggestion
    if (flutter.flavor case final flavor?) {
      effectiveDartDefines['FLUTTER_APP_FLAVOR'] = flavor;
    }
```
To get rid of bang operator

**Review by piotruela — CHANGES_REQUESTED** [review]
Thanks for the contribution!
Overall lgtm, just a small nitpick.
Please also add an entry to changelog.

File: packages/patrol_cli/lib/src/crossplatform/app_options.dart (hunk 1 above)
Comment (GChanathip) [reply packages/patrol_cli/lib/src/crossplatform/app_options.dart]: Thanks for your suggestion.

File: packages/patrol_cli/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 3.5.0
```
Comment (piotruela) [inline packages/patrol_cli/CHANGELOG.md:1]: No need to specify the version just yet. Use `## Unreleased` instead

**Review by piotruela — APPROVED** [review]
All good! 🙌

**Pull request merged**